datafusion/common/src/scalar.rs

```diff
@@ -572,7 +572,12 @@ fn do_date_time_math(
     scalar: &ScalarValue,
     sign: i32,
 ) -> Result<NaiveDateTime> {
-    let prior = NaiveDateTime::from_timestamp(secs, nsecs);
+    let prior = NaiveDateTime::from_timestamp_opt(secs, nsecs).ok_or_else(|| {
```

Comment:
if this fails at runtime, now a real error will be returned rather than panic

Reply:
Not sure if this is the problem elsewhere, we can file an issue tracker

Reply:
I wonder if this is the same as mentioned by @waitingkuo on https://github.com/apache/arrow-datafusion/pull/4156#issuecomment-1311270098 from @andre-cc-natzka 🤔 

Maybe after merging up that PR with this fix we'll get a real error rather than a panic